File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/decorator/dataexport/ExcelDataExportDecorator.java
// ExcelDataExportDecorator.java

package com.bigboxcode.designpattern.decorator.dataexport;

public class ExcelDataExportDecorator extends DataExportDecorator {
    public ExcelDataExportDecorator(DataExport dataExporter) {
        super(dataExporter);
    }

    @Override
    public void processData() {
        dataExporter.processData();
        this.processExcel();
    }

    private void processExcel() {
        System.out.println("Processed data to Excel");
    }
}

File: design-patterns/ts/decorator/data-export/excel-data-export-decorator.ts
// excel-data-export-decorator.ts

import DataExport from "./data-export";
import DataExportDecorator from "./data-export-decorator";

class ExcelDataExportDecorator extends DataExportDecorator {
    constructor(dataExporter: DataExport) {
        super(dataExporter);
    }

    processData(): void {
        this.dataExporter.processData();
        this.processExcel();
    }

    processExcel() {
        console.log("Processed data to Excel");
    }
}

export default ExcelDataExportDecorator;